client/my-sites/stats/stats-purchase/stats-purchase-checkout-redirect.tsx

```diff
@@ -5,20 +5,29 @@ import {
 } from '@automattic/calypso-products';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
 
+const setUrlParam = ( url: URL, paramName: string, paramValue: string ): void => {
```

Comment:
```suggestion
const setUrlParam = ( url: URL, paramName: string, paramValue?: string ): void => {
```

If `paramValue` is optional, let's mark it like this in type definition. If we're just checking whether it's empty - let's check it explicitly in if ... else below.